File: src/main/java/io/github/flegacy/flms/wand/ui/WandMenu.java
package io.github.flegacy.flms.wand.ui;

import io.github.flegacy.flms.ItemLibrary;
import org.jetbrains.annotations.NotNull;

import io.github.flegacy.flms.FLMS;
import io.github.flegacy.flms.ui.FLMSInterface;
import net.kyori.adventure.text.Component;

public class WandMenu extends FLMSInterface {

	public WandMenu(FLMS plugin) {
		super(27, Component.text("FLMS Config Menu"), plugin);
		final ItemLibrary lib = plugin.getItemLibrary();
        
        setElement(10, new TransferButton(lib.getBlockConfigIcon(), null));
		setElement(12, new TransferButton(lib.getToolConfigIcon(), null));
		setElement(14, new TransferButton(lib.getEffectConfigIcon(), this));
		setElement(16, new TransferButton(lib.getRegionConfigIcon(), this));
	}
}

File: src/main/kotlin/io/github/flegacy/flms/config/ui/WandMenu.kt
package io.github.flegacy.flms.config.ui

import io.github.flegacy.flms.FLMS
import io.github.flegacy.flms.items.ItemLibrary
import io.github.flegacy.flms.ui.FLMSInterface
import io.github.flegacy.flms.ui.element.FillerElement
import io.github.flegacy.flms.ui.element.TransferButton
import io.github.flegacy.flms.util.msgFormat
import net.kyori.adventure.text.Component

class WandMenu(plugin: FLMS): FLMSInterface(27, "FLMS Config Menu") {



    init {
        val filler = FillerElement()
        fill(filler)

        val blockConfig = BlockConfigMenu(this, plugin)
        val blockTransfer = TransferButton(ItemLibrary.BLOCK_CONFIG_ICON, blockConfig)
        setElement(11, blockTransfer)
    }


}
